src/Autodesk.Forge.Core/ForgeHandler.cs

```diff
@@ -88,33 +95,34 @@ protected virtual IAsyncPolicy<HttpResponseMessage> GetTokenRefreshPolicy()
                 );
         }
 
-        protected virtual IAsyncPolicy<HttpResponseMessage> GetResiliencyPolicies(int timeoutSeconds = ForgeGatewayTimeoutSeconds)
+        protected virtual IAsyncPolicy<HttpResponseMessage> GetResiliencyPolicies(TimeSpan timeoutValue)
         {
             // Retry when HttpRequestException is thrown (low level network error) or 
             // the server returns an error code that we think is transient
-            var errors = Policy
-                .Handle<HttpRequestException>()
-                .OrResult<HttpResponseMessage>(response =>
-                {
-                    int[] retriable = {
+            //
+            int[] retriable = {
                         (int)HttpStatusCode.RequestTimeout, // 408
                         429, //too many requests
                         (int)HttpStatusCode.BadGateway, // 502
                         (int)HttpStatusCode.ServiceUnavailable, // 503
                         (int)HttpStatusCode.GatewayTimeout // 504
                         };
+            var (retryBaseDelay, retryMultiplier) = GetRetryParameters();
+            var retry = Policy
+                .Handle<HttpRequestException>()
+                .Or<Polly.Timeout.TimeoutRejectedException>()// thrown by Polly's TimeoutPolicy if the inner call times out
+                .OrResult<HttpResponseMessage>(response =>
+                {
                     return retriable.Contains((int)response.StatusCode);
-                });
-
-            // retry 3 times with exponential backoff and jitter while respecting the RetryAfter header from server
-            var retry = errors.WaitAndRetryAsync(
+                })
+                .WaitAndRetryAsync(
```

Comment:
so it means we will do retry based on the response code as well as the exception type? 504 is for the case that DA or Forge cannot response in 10s for some reason, exception is for the case that everything is completely down?

Reply:
Yes, we retry when
1. We get a HttpRequestException. This can happen when some low level networking goes wrong.
2. We get a TimeoutRejectedException. This can happen when a timeout policy causes the request to be cancelled.
3. We get an error code that we think we can retry.